ray-operator/controllers/ray/common/pod.go

```diff
@@ -160,6 +187,40 @@ func BuildPod(podTemplateSpec v1.PodTemplateSpec, rayNodeType rayiov1alpha1.RayN
 
 	setContainerEnvVars(&pod.Spec.Containers[rayContainerIndex], rayNodeType, rayStartParams, svcName)
 
+	// add health check for head node
+
+	// health check only works for 1.13
+	if major, minor, _, err := ParseVersion(rayVersion); err == nil && major*100+minor >= 113 {
+		if rayNodeType == rayiov1alpha1.HeadNode {
+			probe := &v1.Probe{}
+			if pod.Spec.Containers[rayContainerIndex].ReadinessProbe == nil {
+				pod.Spec.Containers[rayContainerIndex].ReadinessProbe = probe
+			}
+			probe = pod.Spec.Containers[rayContainerIndex].ReadinessProbe
+			if probe.Exec == nil && probe.HTTPGet == nil && probe.TCPSocket == nil {
+				probe.Exec = &v1.ExecAction{Command: []string{
+					"python", "-c",
+					"from ray._private.gcs_utils import check_health;import sys;" +
+						fmt.Sprintf("sys.exit(not check_health('localhost:%d'))",
+							DefaultRedisPort),
+				}}
+			}
+			if probe.InitialDelaySeconds == 0 {
+				probe.InitialDelaySeconds = 10
+			}
+			if probe.FailureThreshold == 0 {
+				// This number is set to 20 to make default tests happy
+				probe.FailureThreshold = 20
+			}
+		}
+
+		// add health check for worker node
+		if rayNodeType == rayiov1alpha1.WorkerNode && pod.Spec.Containers[rayContainerIndex].ReadinessProbe == nil {
+			// TODO: add probe for worker node
```

Comment:
Let's not forget to do that 
cc @iycheng @scv119 @wuisawesome 

We should implement `ray health-check --component=raylet`.

Reply:
Or the equivalent http endpoint.

Reply:
yeah, i will make few changes based on recent ray ha discussion with Bruce and Yi

Reply:
code updated